### PR TITLE
fix: AlertReportCronScheduler tests

### PR DESCRIPTION
**Pull request by diegomedina248**
### SUMMARY

Tests for the `AlertReportCronScheduler` component were skipped when the component was rewritten.
This PR introduces tests for the component using the react testing library, plus some optimizations in the component.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
None (other than the test execution results)

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by rusackas** [comment]
@eschutho would love to get feedback from someone closer to this feature, in case there's any risk I'm not spotting here.

File: superset-frontend/src/views/CRUD/alert/components/AlertReportCronScheduler.tsx [hunk 1]
```diff
@@ -16,44 +16,59 @@
...
+    const handleBlur = useCallback(
+      (event: FocusEvent<HTMLInputElement>) => {
+        onChange(event.target.value);
+      },
+      [onChange],
+    );
+
+    const handlePressEnter = useCallback(() => {
+      onChange(inputRef.current?.input.value || '');
+    }, [onChange]);
```
Comment (eschutho) [inline superset-frontend/src/views/CRUD/alert/components/AlertReportCronScheduler.tsx:63]: @diegomedina248 do you think we need `useCallback` for simple Input component props?

**Comment by eschutho** [comment]
> @eschutho would love to get feedback from someone closer to this feature, in case there's any risk I'm not spotting here.

Looks great.. thanks for the tests @diegomedina248!

File: superset-frontend/src/views/CRUD/alert/components/AlertReportCronScheduler.tsx (hunk 1 above)
Comment (diegomedina248) [reply superset-frontend/src/views/CRUD/alert/components/AlertReportCronScheduler.tsx:63]: It doesn't hurt nor improve in this case, but I believe it's cleaner

Comment (eschutho) [reply superset-frontend/src/views/CRUD/alert/components/AlertReportCronScheduler.tsx:63]: It's a bit more code and a bit more computation imo, which is my only concern with not much performance impact, but it's not a big deal. I'll cc @rusackas if he has any opinions.

Comment (eschutho) [reply superset-frontend/src/views/CRUD/alert/components/AlertReportCronScheduler.tsx:63]: but to your comment of it being cleaner, @diegomedina248 I meant to also ask in what way?

**Review by rusackas — APPROVED** [review]

**Pull request merged**